### ISSUE TITLE
Discuss: 'all' JavaScript should not auto initialise, and export individual components

### ISSUE THREAD

**Issue by nickcolley** [alphagov/govuk-frontend#660]
At the moment if you import 'all' it'll initialise automatically.

Should it instead do the following:

```html
<script src="all.js"></script>
<script>
  govukFrontend.initAll()
</script>
```

and

```html
<script src="all.js"></script>
<script>
  govukFrontend.Checkboxes.init()
</script>
```

**Comment by nickcolley** [comment]
Adam has an interesting article on this: https://adamsilver.io/articles/dont-initialise-javascript-automagically/

**Comment by adamsilver** [comment]
Here's something I wrote about initialising behaviour automagically. 

Unfortunately, it's _me_ sharing _my_ article ( :) ) but it documents quite a few of the problems I've faced in the past.

https://adamsilver.io/articles/dont-initialise-javascript-automagically/

I spoke with @nickcolley about potentially have the best of both worlds—sensible defaults, with the ability initialise manually/programmatically, so that's something to consider. I think the current approach allows for that.
